### PR TITLE
Update documentation to include needsdiagnosis model

### PR DESCRIPTION
**Pull request by ksy36**

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -33,6 +33,8 @@ More information on the Mozilla Hacks blog:
 
 - **duplicate** - The aim of this classifier is to detect duplicate bugs.
 
+- **needsdiagnosis** - The aim of this classifier is to detect issues that are likely invalid and don't need to be diagnosed for webcompat usecase.
```
Comment (ksy36) [inline README.md:36]: I've also put together a document describing the whole process https://github.com/webcompat/webcompat.com/blob/main/docs/ml-process.md. @marco-c  wonder if it's useful to add a link to it here ?

Comment (marco-c) [reply README.md:36]: Yes, I think that'd be useful!

Comment (ksy36) [reply README.md]: Ok, added the link :)

File: README.md [hunk 2]
```diff
@@ -125,8 +127,9 @@ Note: the script will take a long time to run (on my laptop more than 7 hours).
 - `bugbug/nlp` contains utility functions for NLP;
 - `bugbug/labels.py` contains utility functions for handling labels;
 - `bugbug/bug_snapshot.py` contains a module to play back the history of a bug.
```
Comment (marco-c) [inline README.md:129]:
```suggestion
- `bugbug/bug_snapshot.py` contains a module to play back the history of a bug;
```

File: README.md [hunk 3]
```diff
@@ -125,8 +127,9 @@ Note: the script will take a long time to run (on my laptop more than 7 hours).
...
+- `bugbug/github.py` contains functions to retrieve issues from GitHub for a specified owner/repository;
```
Comment (marco-c) [inline README.md:130]:
```suggestion
- `bugbug/github.py` contains functions to retrieve issues from GitHub for a specified owner/repository.
```

**Review by marco-c — APPROVED** [review]
Thanks!

**Pull request merged**